codalab/server/worker_info_accessor.py

```diff
@@ -18,21 +19,32 @@ def wrapper(*args, **kwargs):
 
         return wrapper
 
-    def __init__(self, model, timeout_seconds):
+    def __init__(self, model, worker_model, timeout_seconds):
         self._model = model
+        self._worker_model = worker_model
         self._timeout_seconds = timeout_seconds
         self._last_fetch = None
         self._fetch_workers()
 
     def _fetch_workers(self):
-        self._workers = {worker['worker_id']: worker for worker in self._model.get_workers()}
+        self._workers = {worker['worker_id']: worker for worker in self._worker_model.get_workers()}
         self._last_fetch = datetime.datetime.utcnow()
         self._uuid_to_worker = {}
         self._user_id_to_workers = defaultdict(list)
+
         for worker in self._workers.values():
             for uuid in worker['run_uuids']:
                 self._uuid_to_worker[uuid] = worker
-            self._user_id_to_workers[worker['user_id']].append(worker)
+
+            owner_id = worker['user_id']
+            self._user_id_to_workers[owner_id].append(worker)
+
+            # Add the worker to all the users of worker's group except the owner
+            memberships = self._model.batch_get_user_in_group(group_uuid=worker['group_uuid'])
+            for m in memberships:
+                if m['user_id'] != owner_id:
```

Comment:
Probably could make _user_id_to_workers a defaultdict(set) so we wouldn't need this check, but that's a refactoring that's not needed in this PR